### PR TITLE
gh-110383: Clarify "non-integral" wording in pow() docs

### PR DESCRIPTION
**Pull request by adiaholic**
This PR is a fix for an issue listed in #110383.

Use widely used non-integer instead of non-integral.
While both terms can be used to mean the same thing, "non-integer number" is the more precise and commonly used term. "Non-integral number" might sometimes be used in a broader sense in certain mathematical contexts, but it generally carries the same meaning as "non-integer number" when referring to real numbers.

* Issue: gh-110383

### CODE REVIEW

**Comment by davidlowryduda** [comment]
You should rename this to have the issue number in the description. For example, calling it `gh-110383: Address two issues from docs@python.org` would better fit the naming convention.

File: Doc/library/array.rst
Comment (davidlowryduda) [inline Doc/library/array.rst]: I would expect four bytes on almost every typical computing environment, but I wouldn't expect this to necessarily be true on a 32 bit system. This would depend on the C compiler and system. The C standard only guarantees that a signed or unsigned int should occupy at least 16 bits.

For that reason, I don't support this change.

**Comment by erlend-aasland** [comment]
Thanks for the PR! It is preferable to fix only one issue per PR; even one-liners. This is because different changes may have different backporting needs.

File: Doc/library/array.rst
Comment (erlend-aasland) [reply Doc/library/array.rst]: True, @davidlowryduda. OTOH, it could be that CPython required `int` to be at least 32 bits, but I can't find any such requirement in the build system or in any pre-processor guard[^1]. I'm not sure if anyone tried to build and run CPython on a system with 16 bit ints, though.

[^1]: except for the `_datetime` extension module, which explicitly fails to build if `SIZEOF_INT < 4`.

File: Doc/library/functions.rst
Comment (erlend-aasland) [inline Doc/library/functions.rst]: FTR: This sentence was added by @mdickinson in gh-27853 (commit 887a55705bb6c05a507c2886c9978a9e0cff0dd7). A similar change as this PR was made in gh-106984 (commit d55b4da10c56f3299998b5b8fee3a8a744fac76c).

Comment (adiaholic) [reply Doc/library/functions.rst]: I'll wait to get a review from @mdickinson to know what he thinks before making a dedicated PR for this change.

File: Doc/library/array.rst
Comment (erlend-aasland) [reply Doc/library/array.rst]: @adiaholic, can you please revert the change in `array.rst`?

Comment (erlend-aasland) [reply Doc/library/array.rst]: See existing PR:

- #113708

Comment (adiaholic) [reply Doc/library/array.rst]: @erlend-aasland Reverted changes for arrays.rst 👍

File: Doc/library/functions.rst
Comment (mdickinson) [reply Doc/library/functions.rst]: So I agree that `non-integer` is more likely to be immediately meaningful to readers than `non-integral`.

The problem I have with it in this context is that to me (and I suspect I'm not the only one) it's too suggestive of type rather than value. That is, if I'm not thinking too hard then I'd think that of course `2.0` isn't an integer (it's a float!), so the conclusion of the sentence would apply to an exponent of `2.0`. But that's wrong: we want to convey the idea that the _value_ is not an integer, regardless of the type.

Can you suggest an alternative way to word this that makes it clear that "non-integer" applies to `2.3` but not to `2.0`?

Comment (mdickinson) [reply Doc/library/functions.rst]: Maybe:
```
... and an exponent whose value has a nonzero fractional part
```
?

Comment (erlend-aasland) [reply Doc/library/functions.rst]: Thanks for chiming in, Mark! With that in mind, I (FWIW) prefer the existing wording. We can instead consider to expand the example in L1491.

Comment (adiaholic) [reply Doc/library/functions.rst]: I’ll work on putting up an appropriate example then.

Thanks for the input @mdickinson & @erlend-aasland 🙌

**Review by erlend-aasland — APPROVED** [review]
Apart from the leading whitespace in L1565 (IIRC), this looks like an improvement to me. I'll wait for @mdickinson's thumbs up before landing, though.

Thanks!

File: Doc/library/functions.rst
Comment (mdickinson) [inline Doc/library/functions.rst]: nitpick-level, but could we say "float result" instead of "real number", for consistency with the earlier part of this paragraph?

Hmm, but then this isn't quite accurate, since `pow(-9, 2)` returns an `int`, not a `float`. I suppose that one could argue that "real number" covers both cases, but that seems a bit of a stretch.

Comment (mdickinson) [reply Doc/library/functions.rst]: I guess the all-operands `int` case was already covered earlier in the paragraph, though, so "float result" is probably okay here.

Comment (adiaholic) [reply Doc/library/functions.rst]: Thanks for reviewing this, highly appreciated !

**Comment by erlend-aasland** [comment]
ISTM Mark's remark has been addressed; I'll merge this later today :)

**Comment by mdickinson** [comment]
> ISTM Mark's remark has been addressed

Addressed indeed! Thank you.

**Review by mdickinson — APPROVED** [review]

**Comment by erlend-aasland** [comment]
A big thanks to everyone involved :)

**Pull request merged**
